### PR TITLE
Task4 search people refactoring

### PR DESCRIPTION
**Pull request by Platton96**

### CODE REVIEW

File: MoneySplitter.Services/API/SearchApiService.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+namespace MoneySplitter.Services.API
+{
+    public class SearchApiService : ISearchApiService
+    {
+        private readonly IApiUrlBuilder _urlBuilder;
+        private readonly IQueryApiService _queryApiService;
+        private readonly IMapper _maper;
+
+        public SearchApiService(IApiUrlBuilder urlBuilder, IQueryApiService queryApiService, IMapper maper)
```
Comment (Vovanella95) [inline MoneySplitter.Services/API/SearchApiService.cs]: IApiUrlBuilder urlBuilder -> IApiUrlBuilder apiUrlBuilder

File: MoneySplitter.Services/API/SearchApiService.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+            _urlBuilder = urlBuilder;
+            _queryApiService = queryApiService;
+            _maper = maper;
+        }
+
+        public async Task<IEnumerable<UserModel>> SearchUsersAsync(string query)
+        {
+            var searchUsersUri = _urlBuilder.SearchUsers(query);
+
+            var dataUsers = await _queryApiService.GetAsync<IEnumerable<DataUser>>(searchUsersUri);
```
Comment (Vovanella95) [inline MoneySplitter.Services/API/SearchApiService.cs]: rename this to usersData or remove this variable, just make return await _query....

File: MoneySplitter.Win10/Common/SearchEngine.cs [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    {
+        private readonly DispatcherTimer _timer;
+
+        private readonly ISearchApiService _searchApiService;
+        private bool _isSearchInProgress;
+        private string _previousQuery = string.Empty;
+        private string _query;
+        private const int INTERVAL_TIME = 1;
+
+        private TimeSpan TIMER_INTERVAL = TimeSpan.FromSeconds(INTERVAL_TIME);
```
Comment (Vovanella95) [inline MoneySplitter.Win10/Common/SearchEngine.cs]: Don't create consts for another consts. In this case just use TimeSpan.FromSeconds(1);

File: MoneySplitter.Win10/ViewModels/FoundUsersViewModel.cs [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        }
+
+        public void ChangedQuery(string query)
+        {
+            SearchEngine.ChangedQeury(query);
+        }
+
+        protected override void OnActivate()
+        {
+            SearchEngine.Activate();
```
Comment (Vovanella95) [inline MoneySplitter.Win10/ViewModels/FoundUsersViewModel.cs]: use base class method first

base.OnActivate();
SearchEngine.Activate();

File: MoneySplitter.Win10/ViewModels/LoginViewModel.cs [hunk 5]
```diff
@@ -52,9 +52,9 @@ public LoginViewModel( INavigationManager navigationManager, IMembershipService
             }
         }
 
-        public void GoToRegistretion()
+        public void NavigateToRegistretion()
```
Comment (Vovanella95) [inline MoneySplitter.Win10/ViewModels/LoginViewModel.cs]: RegistrAtion

Comment (Vovanella95) [reply MoneySplitter.Win10/ViewModels/LoginViewModel.cs]: Better name to NavigateToRegister

File: MoneySplitter.Win10/ViewModels/FoundUsersViewModel.cs (hunk 4 above)
Comment (Platton96) [reply MoneySplitter.Win10/ViewModels/FoundUsersViewModel.cs]: replace

**Pull request merged**